--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,26 @@
+package(default_visibility = ["//visibility:public"])
+
+static_srcs = glob(["**/*.lib", "**/*.a"], exclude=["**/*jni.lib"])
+shared_srcs = glob(["**/*.dll", "**/*.so*", "**/*.dylib"], exclude=["**/*jni.dll", "**/*jni.so", "**/*.so.debug", "**/libopencv_java*.dylib"])
+shared_jni_srcs = glob(["**/*jni.dll", "**/*jni.so*", "**/*.jni.dylib", "**/libopencv_java*.dylib"], exclude=["**/*.so.debug"])
+
+cc_library(
+    name = "static_libs",
+    srcs = static_srcs,
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "shared_libs",
+    srcs = shared_srcs,
+    visibility = ["//visibility:public"],
+    deps = [
+        ":static_libs",
+    ]
+)
+
+cc_library(
+    name = "shared_jni_libs",
+    srcs = shared_jni_srcs,
+    visibility = ["//visibility:public"],
+)